cmake/kokkos_compiler_id.cmake

```diff
@@ -4,31 +4,34 @@ SET(KOKKOS_CXX_COMPILER ${CMAKE_CXX_COMPILER})
 SET(KOKKOS_CXX_COMPILER_ID ${CMAKE_CXX_COMPILER_ID})
 SET(KOKKOS_CXX_COMPILER_VERSION ${CMAKE_CXX_COMPILER_VERSION})
 
+IF(Kokkos_ENABLE_CUDA)
 # Check if the compiler is nvcc (which really means nvcc_wrapper).
 EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
-                COMMAND grep nvcc
-                COMMAND wc -l
-                OUTPUT_VARIABLE INTERNAL_HAVE_COMPILER_NVCC
+                OUTPUT_VARIABLE INTERNAL_COMPILER_VERSION
                 OUTPUT_STRIP_TRAILING_WHITESPACE)
 
+string(REPLACE "\n" " - " INTERNAL_COMPILER_VERSION_ONE_LINE ${INTERNAL_COMPILER_VERSION} )
+
+string(FIND ${INTERNAL_COMPILER_VERSION_ONE_LINE} "nvcc" INTERNAL_COMPILER_VERSION_CONTAINS_NVCC)
```

Comment:
As much as I prefer lowercase, this is inconsistent with everything else so please `string -> STRING`

Reply:
Is `CMAKE_CXX_COMPILER_VERSION` incorrect here?

Reply:
Done